src/Microsoft.ML.FastTree/TreeEnsembleFeaturizer.cs

```diff
@@ -193,19 +93,50 @@ private void IsNormalized(int iinfo, ref bool dst)
                 InputRoleMappedSchema = schema;
 
                 // A vector containing the output of each tree on a given example.
-                var treeValueType = new VectorType(NumberType.Float, _owner._ensemble.TrainedEnsemble.NumTrees);
+                var treeValueType = new VectorType(NumberType.Float, owner._ensemble.TrainedEnsemble.NumTrees);
                 // An indicator vector with length = the total number of leaves in the ensemble, indicating which leaf the example
                 // ends up in all the trees in the ensemble.
-                var leafIdType = new VectorType(NumberType.Float, _owner._totalLeafCount);
+                var leafIdType = new VectorType(NumberType.Float, owner._totalLeafCount);
                 // An indicator vector with length = the total number of nodes in the ensemble, indicating the nodes on
                 // the paths of the example in all the trees in the ensemble.
                 // The total number of nodes in a binary tree is equal to the number of internal nodes + the number of leaf nodes,
                 // and it is also equal to the number of children of internal nodes (which is 2 * the number of internal nodes)
                 // plus one (since the root node is not a child of any node). So we have #internal + #leaf = 2*(#internal) + 1,
                 // which means that #internal = #leaf - 1.
                 // Therefore, the number of internal nodes in the ensemble is #leaf - #trees.
-                var pathIdType = new VectorType(NumberType.Float, _owner._totalLeafCount - _owner._ensemble.TrainedEnsemble.NumTrees);
-                OutputSchema = Schema.Create(new SchemaImpl(ectx, owner, treeValueType, leafIdType, pathIdType));
+                var pathIdType = new VectorType(NumberType.Float, owner._totalLeafCount - owner._ensemble.TrainedEnsemble.NumTrees);
+
+                // Start creating output schema with types derived above.
+                var schemaBuilder = new SchemaBuilder();
+
+                // Metadata of tree values.
+                var treeIdMetadataBuilder = new MetadataBuilder();
+                ValueGetter<VBuffer<ReadOnlyMemory<char>>> treeIdMetadataGetter = (ref VBuffer<ReadOnlyMemory<char>> value) => owner.GetTreeSlotNames(ref value);
```

Comment:
>(ref VBuffer<ReadOnlyMemory<char>> value) => owner.GetTreeSlotNames(ref value); [](start = 82, length = 79)

Hi @wshin, what is the purpose of this wrapping delegate? Why not just assign `owner.GetTreeSlotNames` directly? It appears, from what I see, to be suitable?

So this:

```csharp
ValueGetter<VBuffer<ReadOnlyMemory<char>>> treeIdMetadataGetter = (ref VBuffer<ReadOnlyMemory<char>> value) => owner.GetTreeSlotNames(ref value);
treeIdMetadataBuilder.Add(MetadataUtils.Kinds.SlotNames, MetadataUtils.GetNamesType(treeValueType.VectorSize), treeIdMetadataGetter);
```

becomes something like this.

```csharp
treeIdMetadataBuilder.Add(MetadataUtils.Kinds.SlotNames, MetadataUtils.GetNamesType(treeValueType.VectorSize), owner.GetTreeSlotNames);
```

Does anything bad happen? Here and the other three places.